daphne_server/src/router/mod.rs

```diff
@@ -99,28 +109,37 @@ impl AxumDapResponse {
             }
         };
 
-        let headers = AppendHeaders([(CONTENT_TYPE, media_type)]);
+        let headers = [(CONTENT_TYPE, media_type)];
 
         Self((StatusCode::OK, headers, response.payload).into_response())
     }
 
     pub fn new_error<E: Into<DapError>>(error: E, metrics: &DaphneServiceMetrics) -> Self {
-        let error = error.into();
-        let status = if matches!(error, DapError::Fatal(_)) {
+        // trigger abort if transition failures reach this point.
+        let error = match error.into() {
+            DapError::Transition(failure) => DapAbort::report_rejected(failure),
+            e @ DapError::Fatal(..) => Err(e),
+            DapError::Abort(abort) => Ok(abort),
+        };
+        let status = if let Err(_e) = &error {
+            // TODO(mendess)
```

Comment:
Probably a good idea to tell your future self what you actually need to do here.

Reply:
I did, it's to uncomment the line below. But I'll make it more obvious